.github/workflows/ci.yaml

```diff
@@ -24,18 +28,6 @@ jobs:
       - name: setup dotnet
         uses: ./.github/dotnet
       - name: dotnet format check
-        run: dotnet format --verify-no-changes *.sln
+        run: dotnet format --verify-no-changes --no-restore *.sln
         env:
           PATH: ${{ github.env.PATH }}:/home/runner/.dotnet/tools
-
-  unit-test:
```

Comment:
removed separate unittest step because it causes a lot duplicate work:
- It builds to whole solution again, while the build itself already does this
- I couldn't get it working to let the test-job depend on the build job build output and use that to run tests.